connexion/operations/abstract.py

```diff
@@ -184,10 +185,20 @@ def _get_val_from_param(self, value, query_defn):
         Convert input parameters into the correct type
         """
 
+    @abc.abstractmethod
+    def _get_val_from_object_param(self, value, query_defn):
+        """
+        Convert object input parameters into the correct type
+        """
+
     def _query_args_helper(self, query_defns, query_arguments,
```

Comment:
I might've overlooked it, but it would be nice to have a separate function which parses query string into a dictionary without the need to create an instance of a class of other dependencies and it's easier to test, e.g

```
def parse_object_str(query_str: str) -> dict
...

# The test would be
assert parse_object_str("id[role]=admin&id[firstName]=Alex") == {"role": "admin", "firstName": "Alex"}

```

Reply:
agreed, but this function takes also non-object style parameters. so the test would actually look like this:

```
def parse_object_str(query_str: str) -> dict
...

# The test would be
assert parse_object_str("foo=bar&id[role]=admin&id[firstName]=Alex") == {"foo": "bar", "id": {"role": "admin", "firstName": "Alex"}}
```

your example was filtered on one key only, which could also make sense, but that requires dissecting the query parameter extra again. i would have one function which takes query parameters, no matter what format, and returns a certain style of values back ...